analyses/bfa/notebooks/bfa_seas_bavg_iriforecast.md

```diff
@@ -0,0 +1,565 @@
+import sys
+import os
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+from src.indicators.drought.config import Config
+
+from src.indicators.drought.iri_rainfallforecast import get_iri_data
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+adm_sel=["Boucle du Mouhoun","Nord","Centre-Nord","Sahel"]
+threshold_mar=40
+threshold_jul=50
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+country="bfa"
+config=Config()
+parameters = config.parameters(country)
+country_data_raw_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,config.RAW_DIR,country)
+country_data_exploration_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,"exploration",country)
+
+stats_reg_path=os.path.join(country_data_exploration_dir,f"{country}_iri_seasonal_forecast_stats_{''.join(adm_sel)}.csv")
+
+adm1_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin1_shp"])
+adm2_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin2_shp"])
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+iri_ds, iri_transform = get_iri_data(config, download=False)
+```
+
+these are the variables of the forecast data, where C indicates the tercile (below-average, normal, or above-average).  
+F indicates the publication month, and L the leadtime
+
+```{code-cell} ipython3
+iri_ds
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+iri_ds.sel(L=1).prob
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+gdf_adm1=gpd.read_file(adm1_bound_path)
+iri_clip=iri_ds.rio.set_spatial_dims(x_dim="lon",y_dim="lat").rio.clip(gdf_adm1.geometry.apply(mapping), iri_ds.rio.crs, all_touched=True)
+```
+
+Below the raw forecast data of below-average rainfall with 1 month leadtime, published in March and July is shown.
+
+```{code-cell} ipython3
+:tags: [hide_input]
+
+g=iri_clip.where(iri_clip.F.dt.month.isin([3]), drop=True).sel(L=1,C=0).prob.plot(
+    col="F",
+    col_wrap=3,
+    cmap=mpl.cm.YlOrRd,
+    cbar_kwargs={
+        "orientation": "horizontal",
+        "shrink": 0.8,
+        "aspect": 40,
+        "pad": 0.1,
+    },
+    figsize=(20,20)
+)
+df_bound = gpd.read_file(adm1_bound_path)
+for ax in g.axes.flat:
+    df_bound.boundary.plot(linewidth=1, ax=ax, color="red")
+    ax.axis("off")
+# fig.tight_layout()
+```
+
+```{code-cell} ipython3
+:tags: [hide_input]
+
+g=iri_clip.where(iri_clip.F.dt.month.isin([7]), drop=True).sel(L=1,C=0).prob.plot(
+    col="F",
+    col_wrap=3,
+    cmap=mpl.cm.YlOrRd,
+    cbar_kwargs={
+        "orientation": "horizontal",
+        "shrink": 0.8,
+        "aspect": 40,
+        "pad": 0.1,
+    },
+    figsize=(20,20)
+)
+df_bound = gpd.read_file(adm1_bound_path)
+for ax in g.axes.flat:
+    df_bound.boundary.plot(linewidth=1, ax=ax, color="red")
+    ax.axis("off")
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+def interpolate_ds(ds,transform,upscale_factor):
+    # Interpolated data
```

Comment:
why is interpolation needed?

Reply:
Because the resolution is very low, which gives skewed results if we would look at the pixels within a region. 
The interpolation method I am using doesn't change the values, but basically splits the pixels into smaller pieces. Do you think that is valid? Could maybe also confirm with IRI